### PR TITLE
[5.0] Prevent resource route root binding

### PR DESCRIPTION
**Pull request by rkgrep**

### CODE REVIEW

File: src/Illuminate/Routing/ResourceRegistrar.php [hunk 1]
```diff
@@ -34,9 +34,15 @@ public function __construct(Router $router)
...
 	 * @param  array   $options
 	 * @return void
+	 *
+	 * @throws \InvalidArgumentException
 	 */
 	public function register($name, $controller, array $options = array())
 	{
+		if ($name == '/')
+		{
+			throw new \InvalidArgumentException('Resource routing is not available for root url');
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/ResourceRegistrar.php]: you need to import this

File: src/Illuminate/Routing/ResourceRegistrar.php [hunk 2]
```diff
@@ -34,9 +34,15 @@ public function __construct(Router $router)
...
+		}
 		// If the resource name contains a slash, we will assume the developer wishes to
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/ResourceRegistrar.php]: leave a new line

File: src/Illuminate/Routing/ResourceRegistrar.php [hunk 3]
```diff
@@ -34,9 +34,15 @@ public function __construct(Router $router)
 	 * @param  string  $controller
 	 * @param  array   $options
 	 * @return void
+	 *
+	 * @throws \InvalidArgumentException
 	 */
 	public function register($name, $controller, array $options = array())
 	{
+		if ($name == '/')
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/ResourceRegistrar.php]: what if name is empty string?

File: tests/Routing/RoutingRouteTest.php [hunk 4]
```diff
@@ -723,6 +723,15 @@ public function testResourceRouting()
 		$this->assertEquals('prefix.foo-bars.show', $routes[0]->getName());
 	}
 
+	/**
```
Comment (GrahamCampbell) [inline tests/Routing/RoutingRouteTest.php]: you need 2 blank lines before the start of the doc

**Comment by rkgrep** [comment]
I've just came to idea that we can allow root mapping for index and create.
And also we can implement extra option to set wildcard and allow root mapping with it.

**Pull request closed without merging**